### DISCUSSION TITLE
Can multiple zenoh networks coexist in same physical network?

### DISCUSSION THREAD

**Discussion by milyin** [eclipse-zenoh/roadmap#19] [Q&A] [answered]
Zenoh nodes can discover each other. But what if someone wants to build distributed application based on Zenoh protocol with guarantee that only instances of this application can be discovered? For example what if, e.g., Somesung and Abble wants to use Zenoh library for interconnecting their home appliances, but do not want to route packets for concurrent? How to achieve that?

**Comment by Mallets** [comment, marked as answer]
Hello Michael,

Zenoh scouting is an optional mechanism that is outside of the core zenoh protocol and it could be implemented in several ways.
One default implementation we provide is a scouting mechanism based on UDP multicast. 

Aa of Today, if the goal is to separate discoverability of different zenoh nodes, it could be achieved by configuring different IP multicast groups (see [config](https://github.com/eclipse-zenoh/zenoh/blob/e155dad547ca7ac5c067574a686656a035b58b6e/EXAMPLE_CONFIG.json5#L109)).
By doing so, zenoh nodes can discover only other application instances in the same IP multicast group.

Zenoh scouting could also be performed via e.g. DNS if the network infrastructure supports it (e.g. connect to "tcp/mysapp.zenoh.local"). 

A new protocol specification is ongoing (see https://github.com/eclipse-zenoh/zenoh/issues/157), one thing we would like to add is to be able to achieve the same kind of UDP multicast scouting behaviour without the need of using different IP multicast groups but rather to support it in the zenoh protocol itself.

In addition to scouting, access control will be needed to whiltelist/blacklist certain connections or zenoh operations. Access control is currently in our [roadmap](https://github.com/orgs/eclipse-zenoh/projects/2) and tracked by https://github.com/eclipse-zenoh/roadmap/issues/8.
